### PR TITLE
fix(window-start): check if window start is a function before executing

### PR DESCRIPTION
**Pull request by Mihoub2**
To resolve the bug that causes the application to crash when building with the message: window.start is not a function.

### CODE REVIEW

**Review by folland87 — COMMENTED** [review]
I think it would be safer to wrap everything in
`if (typeof window !== "undefined")`

**Comment by Mihoub2** [comment]
> I think it would be safer to wrap everything in `if (typeof window !== "undefined")`

Roger!

**Comment by Mihoub2** [comment]
I'm taking this PR opportunity to make a small correction regarding the badges. I've changed the type of the children from "string" to React.ReactNode. Is this okay with you?

**Comment by annelhote** [comment]
> I'm taking this PR opportunity to make a small correction regarding the badges. I've changed the type of the children from "string" to React.ReactNode. Is this okay with you?

Sure but why children is now optional ? IMHO it should be a required param.

File: src/components/Badge/Badge.tsx [hunk 1]
```diff
@@ -6,38 +6,45 @@ import { Link } from "../Link";
...
-  | 'warning'
-  | 'success';
+export type BadgeType = "new" | "error" | "info" | "warning" | "success";
 
 export type BadgeProps = {
   icon?: string;
   className?: Argument;
   size?: "md" | "sm";
-  children: string;
+  children?: React.ReactNode;
```
Comment (annelhote) [inline src/components/Badge/Badge.tsx]: Why optional ?

**Review by annelhote — CHANGES_REQUESTED** [review]

File: src/components/Badge/Badge.tsx (hunk 1 above)
Comment (Mihoub2) [reply src/components/Badge/Badge.tsx]: you are right, it's corrected in another commit

**Review by annelhote — APPROVED** [review]

File: src/hooks/useDSFRConfig.tsx [hunk 2]
```diff
@@ -50,8 +50,13 @@ export const DSFRConfig = ({
       await import("@gouvfr/dsfr/dist/utility/utility.css");
       await import("@gouvfr/dsfr/dist/dsfr.css");
       // @ts-expect-error
-      window.dsfr.start();
```
Comment (folland87) [inline src/hooks/useDSFRConfig.tsx:53]: I think it's better to return early if typeof window === "undefined" in line 39 preferably.
Then here we could just call conditionally
`window.dsfr.start?.()`

Comment (Mihoub2) [reply src/hooks/useDSFRConfig.tsx:53]: check now !

**Pull request merged**